superset-frontend/src/common/components/index.tsx

```diff
@@ -17,16 +17,35 @@
  * under the License.
  */
 import { styled } from '@superset-ui/core';
+// eslint-disable-next-line no-restricted-imports
 import { Skeleton } from 'antd';
+import AntdMenuItem from 'antd/lib/menu/MenuItem';
 
 /*
-  Antd is exported from here so we can override components with Emotion as needed.
+  Antd is re-exported from here so we can override components with Emotion as needed.
 
   For documentation, see https://ant.design/components/overview/
  */
-/* eslint no-restricted-imports: 0 */
+// eslint-disable-next-line no-restricted-imports
+export {
+  Avatar,
+  Card,
+  Collapse,
+  Empty,
+  Dropdown,
+  Menu,
+  Modal,
+  Popover,
+  Skeleton,
+  Tabs,
+  Tooltip,
+} from 'antd';
 
-export * from 'antd';
+export const MenuItem = styled(AntdMenuItem)`
```

Comment:
We could export a patched `Menu` that provides the same compound component support as AntD, so you don't have to change all the other files from `Menu.Item` to `MenuItem`, then people familiar with AntD could use it as they normally would. 

Something like:

```
import { Menu as AntdMenu } from 'antd';

export const Menu = Object.assign(
  AntdMenu,
  {
    Item: styled(AntdMenu.Item)`
      > a {
        text-decoration: none;
      }
    `,
    SubMenu: AntdMenu.SubMenu,
    ItemGroup: AntdMenu. ItemGroup,
    Divider: AntdMenu.Divider,
  },
);
```

Reply:
The above might be good since there are already other PRs in flight leaning on `Menu.Item`, like this one from @kgabryje https://github.com/apache/incubator-superset/pull/11487/files

Reply:
It also means we can similarly style any of the other compound components without having to similarly touch other files.

Reply:
Actually, I didn't look very closely at https://github.com/apache/incubator-superset/pull/11487/files, but it appears that one takes a similar approach to what I'm suggesting above. Should we just add your style on that PR instead?

Reply:
I tried direct assignment and it breaks typing. 

Sure, but I think it's still important to explicitly export `antd` components and potentially make it possible to import other overrides `from "src/common/components"`, otherwise there is no difference between `import { x } from "src/common/components"` than directly importing from `antd`.

Reply:
Ahh, wasn't aware of the typing issues with this approach... I wonder if there's a good way to mitigate that, as I like the ability to just bolt on `styled` parts as needed.

Totally agree with your plan around the explicit antd exports and importing/exporting the other overrides from there.

Reply:
These changes look good to me and it won't be problematic to refactor my PR. However, I'd appreciate if we merged it soon so that I won't have too many conflicts.
What do you mean that it breaks typing? I used `Object.assign` syntax in my PR and there were no typing errors.

Reply:
`Object.assign` seems to be working fine, even for typing. Thanks for the thoughts!